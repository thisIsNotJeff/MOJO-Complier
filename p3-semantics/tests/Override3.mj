type 
  A  = object { a: int; p() };
  AB = A object { b: int };

def Pa(self: A);
def Pab(self: AB);

type T1 = AB object { p := Pab }; /* ok */
type T2 = A object { p := Pa }; /* ok */
type T3 = AB object { p := Pa }; /* ok */
type T4 = A object { p := Pab }; /* error */
