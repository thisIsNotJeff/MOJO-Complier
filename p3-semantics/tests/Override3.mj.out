BEGIN _BUILTIN
_BUILTIN.int=int
_BUILTIN.bool=bool
_BUILTIN.false:bool=false
_BUILTIN.true:bool=true
_BUILTIN.Null=^null
_BUILTIN.nil:Null=0
_BUILTIN.Refany=^null
_BUILTIN.Address=^null
_BUILTIN.Root=null
object {
}
_BUILTIN.Text=^null
_BUILTIN.First:(...)
_BUILTIN.Last:(...)
_BUILTIN.New:(...)
_BUILTIN.Number:(...)
BEGIN Override3
Override3.A=Root
object {
0: a:int
0: p:()=null
}
Override3.AB=A
object {
4: b:int
}
Override3.Pa:(VALUE self:A)
Override3.Pab:(VALUE self:AB)
Override3.T1=AB
object {
4: p:()=Override3.Pab:(VALUE self:AB)
}
Override3.T2=A
object {
4: p:()=Override3.Pa:(VALUE self:A)
}
Override3.T3=AB
object {
4: p:()=Override3.Pa:(VALUE self:A)
}
Override3.T4=A
object {
4: p:()=Override3.Pab:(VALUE self:AB)
}
END Override3
END _BUILTIN
